src/colmap/scene/reconstruction_test.cc

```diff
@@ -49,6 +55,61 @@ bool CheckReconstruction(const Reconstruction& reconstruction) {
   return true;
 }
 
+// Check if two binary files are the same
+bool CompareBinaryFiles(const std::string& file1, const std::string& file2) {
+  std::ifstream f1(file1, std::ios::binary);
+  std::ifstream f2(file2, std::ios::binary);
+
+  if (!f1.is_open() || !f2.is_open()) {
+    LOG(ERROR) << "Error opening one of the files";
+    return false;
+  }
+
+  // Compare file sizes
+  f1.seekg(0, std::ios::end);
+  f2.seekg(0, std::ios::end);
+  std::streampos size1 = f1.tellg();
+  std::streampos size2 = f2.tellg();
+  if (size1 != size2) {
+    return false;
+  }
+
+  // Compare files byte by byte
+  f1.seekg(0, std::ios::beg);
+  f2.seekg(0, std::ios::beg);
+  char byte1, byte2;
+  while (f1.read(&byte1, 1) && f2.read(&byte2, 1)) {
+    if (byte1 != byte2) {
+      return false;  // Files are different
+    }
+  }
+  return true;
+}
+
+// Compare two reconstructions
+// The binary files are compared after serialized to disk
+bool CompareReconstructions(const std::string& test_dir,
```

Comment:
After the new reconstruction_io changes, this can now simply use stringstreams and directly compare the stream contents.

Reply:
Thanks for the PR. This is updated.